src/main/scala/com/azavea/modellab/Parser.scala

```diff
@@ -59,36 +70,75 @@ class Parser(layerRegistry: LayerRegistry, layerReader: FilteringLayerReader[Lay
       LoadLayer(json.param[String]("layer_name"), windowedReader)
 
     case "LocalAdd" => json => {
-      val inputs = json.inputs
-      LocalAdd(inputs(0), inputs(1))
+      LocalBinaryOp(Add, json.inputs, json.optParam[Int]("constant"))
     }
 
-    case "FocalSum" => json => {
-      val inputs = json.inputs
-      val n = Square(1)
-      require(inputs.size == 1, "FocalSum expexects one layer input")
-
-      FocalOp(inputs.head, n, Sum.apply)
-    }
-    
     case "LocalSubtract" => json => {
-      val inputs = json.inputs
-      LocalSubtract(inputs(0), inputs(1))
+      LocalBinaryOp(Subtract, json.inputs, json.optParam[Int]("constant"))
     }
 
     case "LocalMultiply" => json => {
-      val inputs = json.inputs
-      LocalMultiply(inputs(0), inputs(1))
+      LocalBinaryOp(Multiply, json.inputs, json.optParam[Int]("constant"))
     }
 
     case "LocalDivide" => json => {
-      val inputs = json.inputs
-      LocalDivide(inputs(0), inputs(1))
+      LocalBinaryOp(Divide, json.inputs, json.optParam[Int]("constant"))
     }
 
     case "ValueMask" => json => {
       ValueMask(json.inputs.head, json.param[Seq[Int]]("masks"))
     }
+
+    case "Mapping" => json => {
+      val mapFrom = json.param[Seq[Seq[Int]]]("map_from")
+      val mapTo = json.param[Seq[Int]]("map_to")
+      require(mapFrom.size == mapTo.size, "The number of map_from lists must equal the number of map_to values")
+      MappingOp(json.inputs.head, mapFrom, mapTo)
+    }
+
+    case "FocalSum" => json => {
+      val n = Square(json.param[Int]("neighborhood_size"))
+      require(json.inputs.size == 1, "FocalSum expexects one layer input")
+
+      FocalOp(Sum.apply, json.inputs.head, n)
+    }
+
+    case "FocalMax" => json => {
+      val n = Square(json.param[Int]("neighborhood_size"))
+      require(json.inputs.size == 1, "FocalMax expexects one layer input")
+
+      FocalOp(geotrellis.raster.op.focal.Max.apply, json.inputs.head, n)
+    }
+
+    case "FocalMin" => json => {
+      val n = Square(json.param[Int]("neighborhood_size"))
+      require(json.inputs.size == 1, "FocalMin expexects one layer input")
+
+      FocalOp(geotrellis.raster.op.focal.Min.apply, json.inputs.head, n)
+    }
+
+    case "FocalMean" => json => {
+      val n = Square(json.param[Int]("neighborhood_size"))
+      require(json.inputs.size == 1, "FocalMin expexects one layer input")
+
+      FocalOp(geotrellis.raster.op.focal.Mean.apply, json.inputs.head, n)
+    }
+
+    case "FocalAspect" => json => {
+      val n = Square(json.param[Int]("neighborhood_size"))
+      require(json.inputs.size == 1, "FocalMin expexects one layer input")
+
+      AspectOp(Aspect.apply, json.inputs.head, n)
+    }
+
+    case "FocalSlope" => json => {
+      val n = Square(json.param[Int]("neighborhood_size"))
```

Comment:
Can this be further parameterized to include another shape, like Circle, for neighborhood shape?